src/lando/utils/apps.py

```diff
@@ -1,6 +1,46 @@
+import re
+
 from django.apps import AppConfig
+from django.conf import settings
+from django.core.checks import Error, register
+
+from lando.utils.phabricator import PhabricatorAPIException, PhabricatorClient
+
+PHAB_API_KEY_RE = re.compile(r"^api-.{28}$")
 
 
 class UtilsConfig(AppConfig):
     default_auto_field = "django.db.models.BigAutoField"
     name = "lando.utils"
+
+
+@register()
+def phabricator_check(**kwargs):
+    """Check validity of Phabricator settings and check connectivity."""
+    errors = []
+
+    unpriv_key = settings.PHABRICATOR_UNPRIVILEGED_API_KEY
+    priv_key = settings.PHABRICATOR_ADMIN_API_KEY
+
+    if unpriv_key and PHAB_API_KEY_RE.search(unpriv_key) is None:
+        errors.append(
+            "PHABRICATOR_UNPRIVILEGED_API_KEY has the wrong format, "
+            'it must begin with "api-" and be 32 characters long.'
+        )
+
+    if priv_key and PHAB_API_KEY_RE.search(priv_key) is None:
+        errors.append(
+            "PHABRICATOR_ADMIN_API_KEY has the wrong format, "
+            'it must begin with "api-" and be 32 characters long.'
+        )
+
+    if (priv_key or unpriv_key) and not errors:
+        try:
+            PhabricatorClient(
+                settings.PHABRICATOR_URL,
+                settings.PHABRICATOR_UNPRIVILEGED_API_KEY,
```

Comment:
Based on the logic above, in theory `unpriv_key` could be empty and yet we still use `settings.PHABRICATOR_UNPRIVILEGED_API_KEY` unconditionally here. It seems we should only check whichever key is non-`None`, or maybe it's worth checking both to ensure one of the keys isn't expired etc?

Reply:
Good suggestion, this way it is more robust that the legacy checks, and would also help with local setup.